lib/model/database.dart

```diff
@@ -103,7 +104,17 @@ class AppDatabase extends _$AppDatabase {
     );
   }
 
-  Future<int> createAccount(AccountsCompanion values) {
-    return into(accounts).insert(values);
+  Future<int> createAccount(AccountsCompanion values) async {
+    try {
+      return await into(accounts).insert(values);
+    } catch (e) {
+      if (e case SqliteException(
+          extendedResultCode: SqlExtendedError.SQLITE_CONSTRAINT_UNIQUE)) {
```

Comment:
Hmm — this differs from the previous revision in that it looks for a `SqliteException` directly, rather than a `DriftRemoteException` that contains a `SqliteException`. Does this version work when you try it manually in the actual app?

Trying it just now, indeed it doesn't — the exception just propagates uncaught, the same behavior as before the PR.

OTOH this version is what we get in the test environment. It's awkward that the test environment differs from the live app in that way, but I think we can still make it work so that the tests have most of the value they should have:
```suggestion
    } catch (error) {
      final e = (error is DriftRemoteException) ? error.remoteCause : error;
      if (e case SqliteException(
          extendedResultCode: SqlExtendedError.SQLITE_CONSTRAINT_UNIQUE)) {
```

That way just that one line behaves differently in tests vs. the live app, and the rest of the error handling should behave the same.

Reply:
Ack, added a comment to call this out.